.github/workflows/ecosystem-recompile.yml

```
@@ -0,0 +1,43 @@
+name: Ecosystem | Recompile members.json
+
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+
+jobs:
+    runs-on: ubuntu-latest
+    env:
+      tox_env: "py39"
+      python-version: "3.9"
+    steps:
+    # setup deps
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ env.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ env.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
+        pip install -r requirements-dev.txt
+
+    - name: Run tests
+      run: |
+        tox -epy39
+
```

Just add a `git pull` in order to get early push
```suggestion
    - name: Pull modif
      run: git pull
 
```

---

Thanks! Can you explain why this is necessary? I'm not as familiar with GitHub actions and would like to understand

---

Is in a case you have a commit just after the job started. It allow you to (almost) be sure to always have the actual version of the main branch :)
The concurrency help to not be in this situation but in case of a lag, it can always happen and so it's just a security for miss generation

---

Ahh that makes sense – thanks!